graphql_compiler/query_planning/make_query_plan.py

```diff
@@ -66,6 +77,80 @@ class QueryPlanDescriptor:
     output_join_descriptors: List[OutputJoinDescriptor]
 
 
+def _make_simple_execute_node(
+    query_and_parameters: QueryStringWithParameters,
+    schema_info: Union[CommonSchemaInfo, SQLAlchemySchemaInfo],
+    provider_id: str,
+    backend_type: BackendType,
+) -> SimpleExecute:
+    """Generate a SimpleExecuteNode.
+
+    Args:
+        query_and_parameters: the query and parameters for which to create a SimpleExecute.
+        schema_info: schema information to use for query compilation.
+        provider_id: the identifier of the provider to be queried.
+        backend_type: the backend type. Note: if schema_info is of type SQLAlchemySchemaInfo, the
+                      backend_type must be a flavor of SQL.
+
+    Returns:
+        SimpleExecute with all necessary information to execute the given query_and_parameters.
+
+    """
+    # Ensure that if a SQL schema info object is passed, that the backend type is a SQL dialect.
+    if isinstance(schema_info, SQLAlchemySchemaInfo):
+        # Compiled SQL query.
+        compilation_result = compile_graphql_to_sql(schema_info, query_and_parameters.query_string)
+        query = print_sqlalchemy_query_string(compilation_result.query, schema_info.dialect)
+
+        # Determine if the SQL dialect requires post-processing.
+        if backend_type == BackendType.mssql:
+            requires_fold_postprocessing = True
+        elif backend_type == BackendType.postgresql:
+            requires_fold_postprocessing = False
+        else:
+            raise AssertionError(
+                f"Received SQLAlchemySchemaInfo, but a non-SQL backend type {backend_type}."
+            )
+
+        provider_metadata = ProviderMetadata(
+            backend_type=backend_type,
+            requires_fold_postprocessing=requires_fold_postprocessing,
+        )
+    # All other backends use CommonSchemaInfo.
+    else:
+        if backend_type == BackendType.cypher:
+            compilation_result = compile_graphql_to_cypher(
+                schema_info, query_and_parameters.query_string
+            )
+        elif backend_type == BackendType.gremlin:
+            compilation_result = compile_graphql_to_gremlin(
+                schema_info, query_and_parameters.query_string
+            )
+        elif backend_type == BackendType.match:
+            compilation_result = compile_graphql_to_match(
+                schema_info, query_and_parameters.query_string
+            )
+        # TODO(selene): add InterpreterAdapter based backends
```

Comment:
I'm unsure of how to deal with InterpreterAdapters. The [public methods](https://github.com/kensho-technologies/graphql-compiler/blob/interpreted_mode_v3/graphql_compiler/interpreter/api.py#L117) are `interpret_ir` and `interpret_query`. `SimpleExecute` is set up to contain a query string, which is the input of `interpret_query`, but that function also requires the schema to be passed. It seems like to want to use the schema to compile the query in this function, but then our output (the equivalent of the `query` in `SimpleExecute`) would be `IrAndMetadata` rather than a query string. Making the type of `query` be `Union[str, IrAndMetadata]` doesn't really seem to make sense, but maybe it would be more tolerable if we renamed `query` to `backend_input` or something similar? Thoughts?